static/src/components/builder/components/AssistantLLM/AssistantLLMForm.vue

```diff
@@ -0,0 +1,246 @@
+    <div id="content-preview-input">
+      <div>
+        <label class="block-label">
+          Content Preview | <span class="label-tip">Enter text if you want to configure some sample data for content in the component preview.</span>
+          <input
+            :value="contentPreview"
+            type="text"
+            class="form-control form-control-sm"
+            @input="updateContentPreview($event.target.value)"
+          >
+        </label>
+      </div>
+    </div>
+
+    <h4>Model Settings</h4>
+    <div class="form-row">
+      <label class="col-labels">
+        Model | <span class="label-tip">The LLM model to use.</span>
+      </label>
+      <select
+        class="form-control form-control-sm"
+        :value="model"
+        @input="updateModel($event.target.value)"
+      >
+        <option
+          v-for="(v, k) in modelOptions"
+          :key="k"
+          :value="v"
+        >
+          {{ k }}
+        </option>
+      </select>
+    </div>
+    <div class="form-row">
+      <label class="col-labels">
+        Model Parameters | <span class="label-tip">Leave blank to use default value.</span>
+      </label>
+      <textarea
+        style="min-height: 16em;"
+        class="form-control form-control-sm"
+        :value="modelParams"
+        :placeholder="model_params_placeholder"
+        @input="updateModelParams($event.target.value)"
+      />
+    </div>
+    <div class="form-row">
+      <label class="col-labels">
+        Editable | <span class="label-tip">Which of the LLM fields are editable by the worker.</span>
+      </label>
+      <select
+        class="form-control form-control-sm"
+        :value="editable"
+        @input="updateEditable($event.target.value)"
+      >
+        <option
+          v-for="(v, k) in EditableOptions"
+          :key="k"
+          :value="v"
+        >
+          {{ k }}
+        </option>
+      </select>
+    </div>
+    <br>
+    <div class="form-row">
+      <label
+        for="pyb-answer"
+        class="col-labels"
+      >
+        Answer field name | <span class="label-tip">The field where the worker's answer is stored. Can be a JSON path like
+          a.b.c.</span>
+      </label>
+      <input
+        id="pyb-answer"
+        :value="pybAnswer"
+        class="form-control form-control-sm"
+        @input="updatePybAnswer($event.target.value)"
+      >
+    </div>
+  </div>
+</template>
+
+<script>
+import '../../../../../css/component_builder.css';
+import * as types from '../../store/types';
+import { mapMutations, mapState } from 'vuex';
+export default {
+  name: 'AssistantLLMForm',
+  data () {
+    return {
+      model_params_placeholder:
+        `{
+          "max_new_tokens": 16,
+          "temperature": 1.0,
+          "top_k": 3,
+          "top_p": 1.0,
+          "seed": null,
+          "repetition_penalty": 1.0,
+          "num_beams": 1,
+          "logprobs": false
+    }`,
+      modelOptions: { 'FLAN-UL2': 'flan-ul2' },
```

Comment:
Can this be configured as settings supporting future additions not requiring code changes?